src/main/java/hudson/plugins/jira/JiraGlobalConfiguration.java

```diff
@@ -29,5 +29,6 @@ public JiraGlobalConfiguration() {
     @DataBoundSetter
     public void setSites(List<JiraSite> sites) {
         this.sites = sites;
+        save();
```

Comment:
https://github.com/jenkinsci/jenkins/blob/01d7f443c9e3178da7222e1dff3fc8cf82bc6943/core/src/main/java/jenkins/model/GlobalConfiguration.java#L35
As noted by the java doc, you have to call save on the data-bound setters.